static/js/compose_actions.js

```diff
@@ -416,21 +416,19 @@ exports.quote_and_reply = function (opts) {
         // are prone to glitches where you select the
         // text, plus it's a complicated codepath that
         // can have other unintended consequences.)
-        //
-        // Note also that we always put the quoted text
-        // above the current text, which explains us
-        // moving the caret below.  I think this is what
-        // most users will want, and it's consistent with
-        // the behavior we had on FF before this change
-        // (which may have been an accident of
-        // implementation).  If we change this decision,
-        // we'll need to make `insert_syntax_and_focus`
-        // smarter about newlines.
-        textarea.caret(0);
+
+        if (textarea.caret() !== 0) {
+            // Insert a newline before quoted message if there is
+            // already some content in the compose box and quoted
+            // message is not being inserted at the beginning.
+            textarea.caret("\n");
```

Comment:
Reflecting on this, shouldn't we just have this add a `\n` to the start of the content we're going to add?  

I think that might end up being simpler.

Reply:
Yeah, it will be much simpler that way. The only issue is that it will always add a newline before the quote that way even if the compose box was initially empty or the quoted message is being added at the initial position. Although any initial newlines are removed from the message once the message is sent but it does appear while writing the message.

If that's not a problem, we can have the newline to be always added before the content.

Reply:
Well, I was imagining we'd do this conditionally still, just the mechanism would be adding a leading newline rather than doing this `textarea.caret("\n")` here, that generates an off-by-one thing we need to correct elsewhere.

Reply:
Issues with adding a leading newline at the start of `content` instead of here:
* The feedback given the users by adding `[Quoting…]` will get mixed up with the content already present in compose-box (while it should be visible separately).
* While inserting `[Quoting…]` in the compose-box, a leading space may/may not be added to `[Quoting…]` (which ensures that the text being inserted starts with a new word) so it would be difficult to predict where to set the final caret position in the compose-box.

Also, I don't understand the problem with adding a newline here. Can you please explain it to me?